SSL Errors Level Downgrade

When a client cause a SSL error like 
 'CertPathValidatorException'
 'CertificateExpiredException'

```
2018-09-14 10:09:04,226 ERROR: org.graylog2.plugin.inputs.transports.NettyTransport - Error in Input [GELF TCP/5b5addf0a16c995e75a84e45] (channel [id: 0x114d955b, /x.x.x.x:58652 =>
/x.x.x.x:10108])
Caused by: java.security.cert.CertPathValidatorException: validity check failed
        at sun.security.provider.certpath.PKIXMasterCertPathValidator.validate(PKIXMasterCertPathValidator.java:135) ~[?:1.8.0_162]
        at sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:223) ~[?:1.8.0_162]
        at sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:140) ~[?:1.8.0_162]
        at sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(PKIXCertPathValidator.java:79) ~[?:1.8.0_162]
        at java.security.cert.CertPathValidator.validate(CertPathValidator.java:292) ~[?:1.8.0_162]
        at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:357) ~[?:1.8.0_162]
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:270) ~[?:1.8.0_162]
        at sun.security.validator.Validator.validate(Validator.java:260) ~[?:1.8.0_162]
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324) ~[?:1.8.0_162]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:279) ~[?:1.8.0_162]
        at sun.security.ssl.X509TrustManagerImpl.checkClientTrusted(X509TrustManagerImpl.java:130) ~[?:1.8.0_162]
        at sun.security.ssl.ServerHandshaker.clientCertificate(ServerHandshaker.java:1966) ~[?:1.8.0_162]
        at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:237) ~[?:1.8.0_162]
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1052) ~[?:1.8.0_162]
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:992) ~[?:1.8.0_162]
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:989) ~[?:1.8.0_162]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_162]
        at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1467) ~[?:1.8.0_162]
        at org.jboss.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1393) ~[graylog.jar:?]
        at org.jboss.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1256) ~[graylog.jar:?]
        ... 19 more
```
The graylog log file be full very quickly.
So this is a problem because some one with a bad TLS configuration can write 50Go log easy.

Can you change the verbose level to DEBUG ?

* Graylog Version: 2.4.6
* Elasticsearch Version: 5.5.1
* MongoDB Version: 2.6.12
* Operating System: CentOS7
* Browser version: Firefox 60 ESR

--- a/graylog2-server/src/main/java/org/graylog2/inputs/transports/netty/ExceptionLoggingChannelHandler.java
+++ b/graylog2-server/src/main/java/org/graylog2/inputs/transports/netty/ExceptionLoggingChannelHandler.java
@@ -39,13 +39,13 @@ public class ExceptionLoggingChannelHandler extends ChannelInboundHandlerAdapter
                     input.getId(),
                     ctx.channel());
         } else {
-            logger.error("Error in Input [{}/{}] (channel {})",
+            logger.error("Error in Input [{}/{}] (channel {}) (cause {})",
                     input.getName(),
                     input.getId(),
                     ctx.channel(),
                     cause);
         }
 
-        super.exceptionCaught(ctx, cause);
+        ctx.close();
     }
 }

--- a/graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java
+++ b/graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java
@@ -236,6 +236,7 @@ public abstract class AbstractTcpTransport extends NettyTransport {
             handlers.put("tls", getSslHandlerCallable(input));
         }
         handlers.putAll(super.getChildChannelHandlers(input));
+        handlers.put("exception-logger", () -> new ExceptionLoggingChannelHandler(input, LOG));
 
         return handlers;
     }

--- a/graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java
+++ b/graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java
@@ -97,8 +97,17 @@ public abstract class NettyTransport implements Transport {
             @Override
             protected void initChannel(Channel ch) throws Exception {
                 final ChannelPipeline p = ch.pipeline();
+                Map.Entry<String, Callable<? extends ChannelHandler>> postentry = null;
                 for (final Map.Entry<String, Callable<? extends ChannelHandler>> entry : handlerList.entrySet()) {
-                    p.addLast(entry.getKey(), entry.getValue().call());
+                    // Handle exceptions at the top of the (bottom-up evaluated) pipeline
+                    if (entry.getKey().equals("exception-logger")) {
+                        postentry = entry;
+                    } else {
+                        p.addLast(entry.getKey(), entry.getValue().call());
+                    }
+                }
+                if (postentry != null) {
+                    p.addLast(postentry.getKey(), postentry.getValue().call());
                 }
             }
         };

